### DISCUSSION TITLE
DocuSeal behing Nginx Proxy Manager

### DISCUSSION THREAD

**Discussion by fdecourt** [docusealco/docuseal#137] [Q&A]
Hi,
How can I configure NGINX Proxy Manager (NPM) to have the real User IP address and not the NPM IP address ?
Thanks for your help
Regards

**Comment by fdecourt** [comment]
Here is my NGINX Proxy : 

add_header X-Frame-Options "SAMEORIGIN";
add_header Content-Security-Policy "upgrade-insecure-requests";
add_header X-Frame-Options "SAMEORIGIN";
add_header X-XSS-Protection "1; mode=block" always;
add_header X-Content-Type-Options "nosniff" always;
add_header Cache-Control "no-transform" always;
add_header Referrer-Policy no-referrer always;
add_header X-Robots-Tag none;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-Proto $scheme;
proxy_set_header X-Forwarded-Host $host;

I don't know how to make it work, please help ;-)

**Reply by hknobbe** [reply]
Here's my (working) config:

```
location / {
    add_header X-Frame-Options "SAMEORIGIN";
    add_header Content-Security-Policy "upgrade-insecure-requests";
    add_header X-XSS-Protection "1; mode=block" always;
    add_header X-Content-Type-Options "nosniff" always;
    add_header Cache-Control "no-transform" always;
    add_header Referrer-Policy no-referrer always;
    add_header X-Robots-Tag none;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_pass http://$server:$port;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
}
```

I've added it under the advanced settings in the web-ui.

Looks like you had the "add_header X-Frame-Options "SAMEORIGIN";" twice in your config.

**Comment by fdecourt** [comment]
Hi,
Thanks a lot for sharing your config.
I am using this one, but still, in the audit log, the IP address is the docker IP address shown, not the user one !

location / {   
    add_header X-Frame-Options "SAMEORIGIN";
    add_header Content-Security-Policy "upgrade-insecure-requests";
    add_header X-XSS-Protection "1; mode=block" always;
    add_header X-Content-Type-Options "nosniff" always;
    add_header Cache-Control "no-transform" always;
    add_header Referrer-Policy no-referrer always;
    add_header X-Robots-Tag none;
    proxy_pass http://$server:$port;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header X-Forwarded-Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header Host $host;
}

If I use this proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
it will show me to many redirect error

If someone have an other idea ?

Thanks
